src/Parser/TypeParser.php

```diff
@@ -139,26 +139,29 @@ private function parseAtomic(TokenIterator $tokens): Ast\Type\TypeNode
 			$tokens->dropSavePoint(); // because of ConstFetchNode
 		}
 
-		$exception = new ParserException(
-			$tokens->currentTokenValue(),
-			$tokens->currentTokenType(),
-			$tokens->currentTokenOffset(),
-			Lexer::TOKEN_IDENTIFIER
-		);
+		$tokensCopy = $tokens->copy();
+		$exception = static function () use ($tokensCopy): ParserException {
+			return new ParserException(
+				$tokensCopy->currentTokenValue(),
+				$tokensCopy->currentTokenType(),
+				$tokensCopy->currentTokenOffset(),
+				Lexer::TOKEN_IDENTIFIER
+			);
+		};
 
 		if ($this->constExprParser === null) {
-			throw $exception;
+			throw $exception();
 		}
 
 		try {
 			$constExpr = $this->constExprParser->parse($tokens, true);
 			if ($constExpr instanceof Ast\ConstExpr\ConstExprArrayNode) {
-				throw $exception;
+				throw $exception();
 			}
 
 			return new Ast\Type\ConstTypeNode($constExpr);
 		} catch (LogicException $e) {
-			throw $exception;
+			throw $exception();
```

Comment:
It looks like constExprParser no longer throws `LogicException` since https://github.com/phpstan/phpdoc-parser/commit/d8e9fd97ca11f2f24fc1aafbcfb1f78bce762267.